api/src/schemas.py

```diff
@@ -185,12 +186,11 @@ def exists(cls, collection):
 
 
 class S3Input(WorkflowInputBase):
-    # s3 discovery
-    discovery: Literal[Discovery.s3]
-
+    discovery: Literal["s3"]
```

Comment:
nothing against it, but was there something wrong with doing it the way it was done? 😅  i thought it looked cleaner

Reply:
also you could reference it as `Discover.s3` in line 253 instead of "s3"

Reply:
This is just my preference - I don't think we should use an `Enum` for this. The mapping isn't much more readable, and there are potential concerns with adding to enums in the future (if we end up supporting more than s3 and cmr) or switching functionality based on enums (which we do in the `check_sample_files` validator). Would it be enough to use constant variables or `S3_DISCOVERY='s3` for this, or would you rather we stick to the enum?

Reply:
Revisiting this, constant variables will not work due to limitations using `Literal[]`. I can switch it back to the enum implementation, but I do still have concerns with it, some of which I can work around using the solutions discussed [here](https://stackoverflow.com/questions/65209934/pydantic-enum-field-does-not-get-converted-to-string).